azkaban-web-server/src/main/java/azkaban/webapp/servlet/ProjectManagerServlet.java

```diff
@@ -1493,89 +1490,85 @@ private void handleJobPage(final HttpServletRequest req, final HttpServletRespon
-      page.add("pathDelimiter", Constants.PATH_DELIMITER);   
       final Node node = flow.getNode(jobName);
       if (node == null) {
         page.add("errorMsg", "Job " + jobName + " not found.");
         page.render();
         return;
       }
+      page.add("jobid", node.getId());
+      page.add("jobtype", node.getType());
+      if (node.getCondition() != null) {
+        page.add("condition", node.getCondition());
+      }
 
       Props jobProp = this.projectManager
           .getJobOverrideProperty(project, flow, jobName, node.getJobSource());
       if (jobProp == null) {
         jobProp = this.projectManager.getProperties(project, flow, jobName, node.getJobSource());
       }
-
-      page.add("jobid", node.getId());
-      page.add("jobtype", node.getType());
-      if (node.getCondition() != null) {
-        page.add("condition", node.getCondition());
+      final List<Pair<String, String>> jobProperties = new ArrayList<>();
+      for (final String key : jobProp.getKeySet()) {
+        final String value = jobProp.get(key);
+        jobProperties.add(new Pair<>(key, value));
       }
+      page.add("jobProperties", jobProperties);
 
-      final ArrayList<String> dependencies = new ArrayList<>();
+      final List<String> dependencies = new ArrayList<>();
       final Set<Edge> inEdges = flow.getInEdges(node.getId());
       if (inEdges != null) {
         for (final Edge dependency : inEdges) {
           dependencies.add(dependency.getSourceId());
         }
       }
-      if (!dependencies.isEmpty()) {
-        page.add("dependencies", dependencies);
-      }
+      page.add("dependencies", dependencies);
 
-      final ArrayList<String> dependents = new ArrayList<>();
+      final List<String> dependents = new ArrayList<>();
       final Set<Edge> outEdges = flow.getOutEdges(node.getId());
       if (outEdges != null) {
         for (final Edge dependent : outEdges) {
           dependents.add(dependent.getTargetId());
         }
       }
-      if (!dependents.isEmpty()) {
-        page.add("dependents", dependents);
-      }
-
-      // Resolve property dependencies
-      final ArrayList<String> source = new ArrayList<>();
-      final String nodeSource = node.getPropsSource();
-      if (nodeSource != null) {
-        source.add(nodeSource);
-        FlowProps parent = flow.getFlowProps(nodeSource);
-        while (parent.getInheritedSource() != null) {
-          source.add(parent.getInheritedSource());
-          parent = flow.getFlowProps(parent.getInheritedSource());
+      page.add("dependents", dependents);
+
+      // Resolve inherited properties
+      final List<Pair<String, String>> allParentFlows = flow.getParents();
+      // List of triplets of NAME and NODE PATH of flows from which properties are
+      // inherited as well as the FILE NAME where they are to be found
+      final List<String[]> inheritedProperties = new ArrayList<>();
```

Comment:
This is a bug prone logic. Can you please make a private class with such triplets?